src/app/components/weather-main-component/weather-main.component.ts

```diff
@@ -88,19 +91,31 @@ export class WeatherMainComponent implements OnInit, OnDestroy, OnChanges {
         for (let i = 0; i < this.cityDetails.length; i++) {
             if (this.cityDetails[i].id === id) {
                 this.cityDetails.splice(i, 1);
-                tempNumOfPages = Math.ceil(this.cityDetails.length / CONSTS.ITEMS_IN_PAGE);
+                tempNumOfPages = Math.ceil(this.cityDetails.length / VIEW_MODEL_CONFIG.citieslimit);
                 if (tempNumOfPages !== this.numOfPages) {
                     this.pagesArr = this.arrayOfPages(tempNumOfPages);
                 }
             }
         }
     }
 
+    updateViewModel(viewModel: viewModel) {
+        this.viewmodel = viewModel;
+        this.viewmodel.tempkey = '2' + this.viewmodel.tempkey.toUpperCase();
+        if (viewModel.citieslimit) {
+            //reset pages
+            this.pageStartItemIndex = 0;
+            this.pagesFinishItemIndex = viewModel.citieslimit;
+            this.numOfPages = Math.ceil(this.cityDetails.length / viewModel.citieslimit);
+            this.pagesArr = this.arrayOfPages(this.numOfPages);
+        }
+    }
+
     addCityInfo(cityInfo: cityDetails) {
         let tempNumOfPages: number;
         cityInfo.dt = Date.now() / 1000;
         this.cityDetails.unshift(cityInfo);
-        tempNumOfPages = Math.ceil(this.cityDetails.length / CONSTS.ITEMS_IN_PAGE);
+        tempNumOfPages = Math.ceil(this.cityDetails.length / VIEW_MODEL_CONFIG.citieslimit);
```

Comment:
This logic is duplicated few times in this files.  I would be better to place it in one point. 
Even now it causes a bug. 
Try to set 49 cities on page an then try to navigate between pages.